etcdctlv3/command/txn_command.go

```diff
@@ -41,198 +51,178 @@ func txnCommandFunc(cmd *cobra.Command, args []string) {
-	if err != nil {
-		ExitWithError(ExitInvalidInput, err)
+func parseRequestUnion(line string) (*clientv3.Op, error) {
+	args := argify(line)
+	if len(args) < 2 {
+		return nil, fmt.Errorf("invalid txn compare request: %s", line)
 	}
 
-	if len(line) == 1 {
-		return failureState
-	}
+	opc := make(chan clientv3.Op, 1)
 
-	// remove trialling \n
-	line = line[:len(line)-1]
-	ru, err := parseRequestUnion(line)
-	if err != nil {
-		ExitWithError(ExitInvalidInput, err)
+	put := NewPutCommand()
+	put.Run = func(cmd *cobra.Command, args []string) {
+		key, value, opts := getPutOp(cmd, args)
+		opc <- clientv3.OpPut(key, value, opts...)
 	}
-
-	txn.Success = append(txn.Success, ru)
-
-	return successState
-}
-
-func failureState(txn *pb.TxnRequest, r *bufio.Reader) stateFunc {
-	fmt.Println("entry failure request[method key value(end_range)] (end with empty line):")
-
-	line, err := r.ReadString('\n')
-	if err != nil {
-		ExitWithError(ExitInvalidInput, err)
+	get := NewGetCommand()
+	get.Run = func(cmd *cobra.Command, args []string) {
+		key, opts := getGetOp(cmd, args)
+		opc <- clientv3.OpGet(key, opts...)
 	}
-
-	if len(line) == 1 {
-		return nil
+	del := NewDelCommand()
+	del.Run = func(cmd *cobra.Command, args []string) {
+		key, opts := getDelOp(cmd, args)
+		opc <- clientv3.OpDelete(key, opts...)
 	}
+	cmds := &cobra.Command{SilenceErrors: true}
+	cmds.AddCommand(put, get, del)
 
-	// remove trialling \n
-	line = line[:len(line)-1]
-	ru, err := parseRequestUnion(line)
-	if err != nil {
-		ExitWithError(ExitInvalidInput, err)
+	cmds.SetArgs(args)
+	if err := cmds.Execute(); err != nil {
+		return nil, fmt.Errorf("invalid txn request: %s", line)
 	}
 
-	txn.Failure = append(txn.Failure, ru)
-
-	return failureState
+	op := <-opc
+	return &op, nil
 }
 
-func parseRequestUnion(line string) (*pb.RequestUnion, error) {
+func parseCompare(line string) (*clientv3.Cmp, error) {
 	parts := strings.Split(line, " ")
-	if len(parts) < 2 {
+	if len(parts) != 4 {
 		return nil, fmt.Errorf("invalid txn compare request: %s", line)
 	}
 
-	ru := &pb.RequestUnion{}
-	key := []byte(parts[1])
-	switch parts[0] {
-	case "r", "range":
-		if len(parts) == 3 {
-			ru.Request = &pb.RequestUnion_RequestRange{
-				RequestRange: &pb.RangeRequest{
-					Key:      key,
-					RangeEnd: []byte(parts[2]),
-				}}
-		} else {
-			ru.Request = &pb.RequestUnion_RequestRange{
-				RequestRange: &pb.RangeRequest{
-					Key: key,
-				}}
-		}
-	case "p", "put":
-		ru.Request = &pb.RequestUnion_RequestPut{
-			RequestPut: &pb.PutRequest{
-				Key:   key,
-				Value: []byte(parts[2]),
-			}}
-	case "d", "deleteRange":
-		if len(parts) == 3 {
-			ru.Request = &pb.RequestUnion_RequestDeleteRange{
-				RequestDeleteRange: &pb.DeleteRangeRequest{
-					Key:      key,
-					RangeEnd: []byte(parts[2]),
-				}}
-		} else {
-			ru.Request = &pb.RequestUnion_RequestDeleteRange{
-				RequestDeleteRange: &pb.DeleteRangeRequest{
-					Key: key,
-				}}
-		}
+	cmpType := ""
+	switch parts[2] {
```

Comment:
this should be part[1]?

bnf is <key> <CMPOP> <(CMPCREATE|CMPMOD|CMPVAL|CMPVER)>